### PR TITLE
feat: add additional traces table filter

### PR DESCRIPTION
**Pull request by maxdeichmann**
> [!IMPORTANT]
> Add Clickhouse support for filtering traces and scores with schema updates and client-side integration.
> 
>   - **Database Changes**:
>     - Add `queue_id` column to `scores` table in Clickhouse.
>   - **Server-Side Logic**:
>     - Add `randomCharacters()` function in `clickhouse-filter.ts` for unique variable names.
>     - Update `createFilterFromFilterState()` in `factory.ts` to use new column mappings.
>     - Modify `getTracesTable()` in `traces.ts` to apply new filters.
>   - **Client-Side Components**:
>     - Update `tracesTableUiColumnDefinitions` in `mapTracesTable.ts` to use `clickhouseSelect` and `queryPrefix`.
>   - **Miscellaneous**:
>     - Remove unused column validation logic in `factory.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 4367cf1b6b3ced6cb6d238897cb5650216cd638e. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts [hunk 1]
```diff
@@ -84,9 +96,10 @@ export class NumberFilter implements Filter {
   }
 
   apply(): ClickhouseFilter {
-    const varName = `stringFilter${this.field}`;
+    const uid = randomCharacters();
```
Comment (Steffen911) [inline packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:99]: Would `crypto.randomBytes(4).toString('hex')` work as well? You don't need the full set of chars, right?
Here, you'd get a string like `c9a6fff4`.

File: packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts [hunk 2]
```diff
@@ -241,11 +258,12 @@ export class FilterList {
...
-    const queries = this.filters.map((filter) => filter.apply().query);
-    const params = this.filters.reduce((acc, filter) => {
-      return { ...acc, ...filter.apply().params };
-    }, {});
+    const compiledQueries = this.filters.map((filter) => filter.apply());
+    const params = compiledQueries
+      .map((q) => q.params)
+      .reduce((acc, val) => ({ ...acc, ...val }), {});
 
+    const queries = compiledQueries.map((q) => q.query);
```
Comment (Steffen911) [inline packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:266]:
```typescript
const { queries, params } = this.filters.map((filter) => filter.apply()).reduce((acc, val) => ({
 queries: [...acc.queries, val.query],
 params: {...acc.params, val.params},
})), { queries: [], params: {} });
```
May also work and make it clear how the parameters relate.

File: packages/shared/src/tableDefinitions/mapTracesTable.ts [hunk 3]
```diff
@@ -7,96 +7,92 @@ export const tracesTableUiColumnDefinitions: UiColumnMapping[] = [
...
     uiTableName: "Tags",
     clickhouseTableName: "traces",
-    clickhouseColumnName: "tags",
+    clickhouseSelect: "tags",
   },
   {
     uiTableName: "Input Tokens",
     clickhouseTableName: "traces",
-    clickhouseColumnName: "promptTokens",
+    clickhouseSelect: "arrayElement(usage_details, 'input')",
```
Comment (Steffen911) [inline packages/shared/src/tableDefinitions/mapTracesTable.ts:66]: arrayElement is a bit confusing here. Isn't usage_details a map?

File: packages/shared/src/tableDefinitions/mapTracesTable.ts [hunk 4]
```diff
@@ -7,96 +7,92 @@ export const tracesTableUiColumnDefinitions: UiColumnMapping[] = [
...
-    uiTableName: "Usage",
-    clickhouseTableName: "traces",
-    clickhouseColumnName: "totalTokens",
+    clickhouseSelect: "arrayElement(usage_details, 'total')",
   },
   {
     uiTableName: "Latency (s)",
     clickhouseTableName: "traces",
-    clickhouseColumnName: "latency",
+    clickhouseSelect: "latency",
```
Comment (Steffen911) [inline packages/shared/src/tableDefinitions/mapTracesTable.ts:81]: Is this a computed column or where would this come from?

**Review by Steffen911 — APPROVED** [review]

File: packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts (hunk 1 above)
Comment (maxdeichmann) [reply packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:99]: I would want to have characters only without numbers.

Comment (Steffen911) [reply packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:99]: @maxdeichmann Ehm, but you have numbers in your code snippet above? 😄

Comment (maxdeichmann) [reply packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:99]: removed. thx.

**Pull request merged**